In [1]:
from unityagents import UnityEnvironment
import numpy as np
import random
from collections import deque

import torch
import torch.nn.functional as F
import torch.optim as optim
from model import QNetwork

import matplotlib.pyplot as plt
%matplotlib inline 

In [2]:
env = UnityEnvironment(file_name="/Users/nathan/Documents/Learning/Udacity_deepRL/Value-based-methods/p1_navigation/Banana.app")

INFO:unityagents:
'Academy' started successfully!
Unity Academy name: Academy
        Number of Brains: 1
        Number of External Brains : 1
        Lesson number : 0
        Reset Parameters :
		
Unity brain name: BananaBrain
        Number of Visual Observations (per agent): 0
        Vector Observation space type: continuous
        Vector Observation space size (per agent): 37
        Number of stacked Vector Observation: 1
        Vector Action space type: discrete
        Vector Action space size (per agent): 4
        Vector Action descriptions: , , , 


In [4]:
from agent import DemoAgent

agent = DemoAgent(37, 4, 0, 'checkpoint_magic-violet-19.pth')

brain_name = env.brain_names[0] 
brain = env.brains[brain_name] 
env_info = env.reset(train_mode=True)[brain_name] # Train mode is fast moving

num_episodes = 500 
max_timesteps = 500 
scores = []
scores_window = deque(maxlen=100)

for episode_num in range(1, num_episodes+1):
    score = 0
    # reset environment 
    env_info = env.reset(train_mode=True)[brain_name]
    
    for timestep in range(max_timesteps):
        # Choose an action based on an inference 
        state = env_info.vector_observations[0]
        action = agent.act(state) 

        # Apply action to the environment and get the results 
        env_info = env.step(action)[brain_name]
        next_state = env_info.vector_observations[0]   # get the next state
        reward = env_info.rewards[0]                   # get the reward
        done = env_info.local_done[0]                   

        state = next_state
        score += reward      
        if done:
            break    
    
    scores_window.append(score)  # Save the most recent score 
    scores.append(score) 
    print('\rEpisode {}\tAverage Score: {:.2f}'.format(episode_num, np.mean(scores_window)), end="")
    if episode_num % 100 == 0:
        print('\rEpisode {}\tAverage Score: {:.2f}'.format(episode_num, np.mean(scores_window)))

env.close() 

Episode 100	Average Score: 16.89
Episode 200	Average Score: 15.78
Episode 300	Average Score: 15.77
Episode 400	Average Score: 16.38
Episode 500	Average Score: 15.32


neat-fog-17
    Episode 100	Average Score: 13.14
    Episode 200	Average Score: 13.55
    Episode 300	Average Score: 14.11
    Episode 400	Average Score: 13.93
    Episode 500	Average Score: 13.89
    
magic-violet-19
    Episode 100	Average Score: 16.89
    Episode 200	Average Score: 15.78
    Episode 300	Average Score: 15.77
    Episode 400	Average Score: 16.38
    Episode 500	Average Score: 15.32
    